src/main/java/org/embulk/input/marketo/rest/MarketoBaseRestClient.java

```diff
@@ -160,7 +162,39 @@ protected boolean isExceptionToRetry(Exception exception)
 
     protected <T> T doPost(final String target, final Map<String, String> headers, final Multimap<String, String> params, Jetty92ResponseReader<T> responseReader, final ContentProvider content)
     {
-        return doRequest(target, HttpMethod.POST, headers, params, content, responseReader);
+        return doRequestWithWrappedException(target, HttpMethod.POST, headers, params, content, responseReader);
+    }
+
+    private String getAccessTokenWithWrappedException()
```

Comment:
Why dont add this try catch when call "response = retryHelper.requestWithRetry" instead of create new function getAccessTokenWithWrappedException

Reply:
It will look cleaner. If try, catch in each block you will duplicate the code